### PR TITLE
feat: support relative paths for xo.path setting (closes #106)

### LINKED ISSUE #105: Allow using relative paths for settings
There’s two possible ways to go about this:

- Allow `./blah` relative paths in addition to `file://` URLs, or
- Implement variable substitution (`${workspaceFolder}`): Since inexplicably VS Code doesn’t export the mechanism it uses itself (microsoft/vscode#2809), we could use a subset of [the code PythonVSCode uses](https://github.com/DonJayamanne/pythonVSCode/blob/e8b109c2f21e56128620abd881e04fded9b7fd76/src/client/common/variables/systemVariables.ts)

One step in the resolution of #71

### PR DESCRIPTION
**Pull request by flying-sheep**
Untested so far

Fixes #105

### CODE REVIEW

File: server/server.js [hunk 1]
```diff
@@ -460,11 +461,15 @@ class Linter {
...
-		if (isSANB(customUri)) {
-			xoUri = customUri;
-		} else {
+		if (!isSANB(customPath)) {
 			xoFilePath = await Files.resolve('xo', undefined, folderPath);
 			xoUri = URI.file(xoFilePath).toString();
+		} else if (customPath.startsWith('file://') {
+			xoUri = customUri;
+		} else if (customPath.startsWith('./')) {
+			xoUri = URI.file(path.join(folderPath, customPath)).toString();
```
Comment (flying-sheep) [inline server/server.js:470]: `folderPath` is the workspace path, right?

**Comment by flying-sheep** [comment]
ah, I wasn’t ready! I thought I had done a draft PR, but apparently not. Here’s the remaining commits: #107

**Comment by Spence-S** [comment]
@flying-sheep no worries! I jumped the gun - gonna fix and publish on my end. Thanks for your work!

**Comment by flying-sheep** [comment]
great! It seems like it works, and loads `xo` from the specified path!

**Pull request merged**